enhancements/run-bundle.md

```diff
@@ -0,0 +1,263 @@
+- [X] Design details are appropriately documented from clear requirements
+- [X] Test plan is defined
+- [X] Graduation criteria for dev preview, tech preview, GA
+
+## Open Questions
+
+1. We currently check OLM installation status with `operator-sdk olm status`.
+   Can we change that to using the discovery API?
+   * Based on discussions, using the discovery API is the way to go.
+
+2. How do we handle --mode= flag on registry add
+   * should we ALWAYS default to --mode=semver?
+     * this has less testing and isnot yet the default. We should stick with
+        the default.
+   * should we have a flag that allows our default mode handling to be overridden?
+     * no
+
+## Summary
+
+Enhance the Operator SDK `run` command to support the new bundle format.
+
+## Motivation
+
+One of the primary goals of the Operator SDK is to make it easy for operator
+developers to run and test their operators with Operator Lifecycle
+Manager (OLM). There currently exists a `run --olm` command which supports the
+packagemanifest format. In order to add support for the new bundle format, we
+are migrating to subcommands on the run command instead of increasing the
+number of flags.
+
+### Goals
+
+Provide a command line that works with bundle image format.
+
+### Non-Goals
+
+This design does not cover the replacement of run --olm with run
+packagemanifest. The change to `run packagemanifest` is covered by
+[OSDK-1126](https://issues.redhat.com/browse/OSDK-1126).
+
+## Proposal
+
+### Implementation Details/Notes/Constraints [optional]
+
+The Operator SDK currently supports running an operator with OLM using the
+`run` command with the `--olm` flag. The existing command only supports the
+package manifest format. This enhancement will add a new subcommand, `bundle`,
+to the `run` command. This new command, `run bundle`, will support the new
+bundle format. More information about the different formats can be found at
+the URLs below:
+
+ * Operator bundles: https://github.com/operator-framework/operator-registry/blob/master/docs/design/operator-bundle.md
+ * Package manifests format: https://github.com/operator-framework/operator-registry/tree/v1.5.3/#manifest-format
+
+The idea is to consolidate the run command into different subcommands to make
+things easier to understand and eliminate flag bloat.
+
+ * `run local` - exists today
+ * `run packagemanifest` - covered by [OSDK-1126](https://issues.redhat.com/browse/OSDK-1126)
+ * `run bundle` - covered by this design document
+
+#### Command Line Interface
+The new command line will look as follows:
+
+```
+operator-sdk run bundle <bundle-image> [--index-image=] [--namespace=] [--install-mode=(AllNamespace|OwnNamespace|SingleNamespace=)]
+```
+
+* `<bundle-image>` is a positional argument that specifies the bundle image.
+* `[--index-image=]` is an optional flag that specifies the index image.
+  * If `--index-image` is not provided, default to
+    `quay.io/operator-framework/upstream-registry-builder`
```

Comment:
I had thought that the intention was to move to use the `upstream-opm-builder` for this?